codec/reflectcodec/type_codec.go

```diff
@@ -332,6 +349,30 @@ func (c *genericCodec) marshal(value reflect.Value, p *wrappers.Packer, maxSlice
 				return err
 			}
 		}
+		return nil
+	case reflect.Map:
+		keys := value.MapKeys()
```

Comment:
I don't think these are guaranteed to be sorted, right?

Reply:
For the `size` method I think it's fine to skip sorting keys. But for the `marshal` we should be sorting those keys, as there is no guarantee that keys will be ordered in maps.

Reply:
> I don't think these are guaranteed to be sorted, right?
That's a really good catch. I don't think they are guarantee. I'll sort it here.

Reply:
Sorting has been added in 9a2876df0 @ceyonur, can you please take a look?